pkg/reconciler/channel/channel.go

```diff
@@ -96,9 +113,129 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, c *v1.Channel) pkgreconc
 		c.Status.MarkDeadLetterSinkNotConfigured()
 	}
 
+	c.Status.Policies = nil
+	applyingEvenPolicies, err := auth.GetEventPoliciesForResource(r.eventPolicyLister, v1.SchemeGroupVersion.WithKind("Channel"), c.ObjectMeta)
+	if err != nil {
+		logging.FromContext(ctx).Errorw("Unable to get applying event policies for Channel", zap.Error(err))
+		c.Status.MarkEventPoliciesFailed("EventPoliciesGetFailed", "Failed to get applying event policies")
+	}
+
+	if len(applyingEvenPolicies) > 0 {
+		unreadyEventPolicies := []string{}
+		for _, policy := range applyingEvenPolicies {
+			if !policy.Status.IsReady() {
+				unreadyEventPolicies = append(unreadyEventPolicies, policy.Name)
+			} else {
+				// only add Ready policies to the list
+				c.Status.Policies = append(c.Status.Policies, eventingduckv1.AppliedEventPolicyRef{
+					Name:       policy.Name,
+					APIVersion: eventingv1alpha1.SchemeGroupVersion.String(),
+				})
+			}
+		}
+
+		if len(unreadyEventPolicies) == 0 {
+			c.Status.MarkEventPoliciesTrue()
+		} else {
+			c.Status.MarkEventPoliciesFailed("EventPoliciesNotReady", "event policies %s are not ready", strings.Join(unreadyEventPolicies, ", "))
+		}
+	} else {
+		// we have no applying event policy. So we set the EP condition to True
+		if featureFlags.IsOIDCAuthentication() {
+			// in case of OIDC auth, we also set the message with the default authorization mode
+			c.Status.MarkEventPoliciesTrueWithReason("DefaultAuthorizationMode", "Default authz mode is %q", featureFlags[feature.AuthorizationDefaultMode])
+		} else {
+			// in case OIDC is disabled, we set EP condition to true too, but give some message that authz (EPs) require OIDC
+			c.Status.MarkEventPoliciesTrueWithReason("OIDCDisabled", "Feature %q must be enabled to support Authorization", feature.OIDCAuthentication)
```

Comment:
I feel like here we want "false" ?

Reply:
This is the case, when no EP for the resource exists.
In case we have OIDC disabled, but an EP exists, the EP will be "NotReady". Then all the EventPolicyConditions on resources for which the EP applies will become False too